aten/src/ATen/Dispatch.h

```diff
@@ -69,3 +69,21 @@
             "%s not implemented for '%s'", (NAME), the_type.toString()); \
     }                                                                    \
   }()
+
+#define AT_DISPATCH_ALL_MATH_TYPES(TYPE, NAME, ...)                      \
```

Comment:
Ugh, how are `ALL_TYPES_AND_HALF` not math types? How did we write ops for `half` if we didn't have this macro?

Reply:
I think the pattern Peter was using is to:

```

#include "ATen/cuda/CUDATypeConversion.cuh"
...
using cuda_scalar_t = cuda::type<scalar_t>;
```

Where `scalar_t` is `Half` and `cuda_scalar_t` becomes `half`